Armadillo/src/main/java/com/scribd/armadillo/playback/MediaSessionCallback.kt

```diff
@@ -184,8 +185,20 @@ internal class MediaSessionCallback(private val onMediaSessionEventListener: OnM
     }
 
     override fun onSeekTo(posInMilis: Long) {
-        playbackEngine?.seekTo(posInMilis.milliseconds)
-        Log.v(TAG, "onSeekTo: $posInMilis")
+        // if the shift has been added, then it must have originated from app UI, because we added it
+        val absolutePosition = if (posInMilis >= AUDIO_POSITION_SHIFT_IN_MS) {
+            posInMilis - AUDIO_POSITION_SHIFT_IN_MS // undo
+        } else {
+            // possibly from notification which sends position relative to chapter start
+            // so, add chapter start time to make it absolute position
```

Comment:
Wouldn't updating notification position offset be a better fix?

Reply:
Not sure if I understood correctly. We don't have any control with notification as such, other than the metadata, session token, etc. I wish there was an option to specify the current state information such as offset when we build notification but there are none.